.travis.yml

```diff
@@ -10,7 +10,7 @@ notifications:
 os:
   - linux
 rust:
-  - 1.34.2
+  - 1.35.0
```

Comment:
Oh darn. Given that `zerocopy` only requires the feature of `1.35` for **two** lines of code that we do not make use of.
See: https://fuchsia.googlesource.com/fuchsia/+/master/garnet/public/rust/zerocopy/src/lib.rs#1065

Reply:
Yeah I'd prefer to go to a lower version of zerocopy instead if possible, but haven't had time to go through their history since there's no changelog. The compiler bump here is just for CI testing the rest of the PR, unless doing a one version bump wouldn't be a dealbreaker.

Reply:
There is none, I think. It was introduced somewhere around `0.2.2` for sure, I've waited on that feature being stabilized [in `image-canvas`](https://github.com/image-rs/canvas/issues/1). And note that all version prior to `0.2.4` have an unsound interface.

Reply:
Also, this would entail restricting the version with an upper bound since `cargo` does not select crates based on compiler requirements (yet). So in essence, we would require an unsound version.. Not the route we want to go probably.

Reply:
So, this leaves us with two choices:
* Bump the compiler version. This is somewhat contrary to making `this as noncontroversial as possible to merge` but honestly, it is a very large and useful feature and the PR is against the major version branch so it is a real possibility :+1: 
* Use something other than `zerocopy`. In the long term it is maybe the most sound solution available but this PR uses only a very small amount of its features so another crate (or porting the code in question, note that this route would *require* a sealed trait imo) could be a better fit.

Reply:
Replacing zerocopy with code in the crate would be easy, < 4 lines of unsafe that's clearly safe as it's casting between integer slice types, but given how close image is to being unsafe-free it would be nice to use zerocopy or another similarly-proven alternative until when/if safe transmutes arrive.

If zerocopy would be useful elsewhere in the crate that could be an argument in its favor. @fintelia mentioned this for pixel structs, which could derive `AsBytes` if `Primitive` had an `AsBytes` bound. This might even get at removing the last two `unsafe`s.

For Rust version bumping, usually my opinion would be that it's a one week bump in the version number five months since image's last MSRV bump, so seems fine, but I also see the motive for staying with 1.34.2 in particular because its approximate to a LTS version being in Debian buster.

Reply:
This already matches my own thoughts closely, the PR makes use of `&[u16] -> &[u8]` conversion (trivial `unsafe`) and `&[u8] -> Option<&[u16]>` (not quite trivial but also not incredibly complex). There are no preconditions for the first conversion and the alignment checks that need to be done at runtime for the second case are clear.

With regards to safety, I don't think it matters particularly if `zerocopy` provides these or we include them here. First of all it is a small finite set, *none depend on type parameters* and the publicly exposed interface of it is rather minimal. In a way, the (total) unsafety introduced by depending on `zerocopy` is greater since these are not true for that crate!

The *potential* introduction into `std` (although similar thoughts had been voiced in previous years and still nothing concrete has happened) also points me in the direction of not bumping the MSRV because we would certainly prefer using the standard utilities over `zerocopy` where possible. Using `zerocopy` for small benefit now means we'll be doing another one soon (compatibility with `1.24.1` was kept over ~1.5 years).

I lean towards implementing the conversions in `utils`, but this conclusion and reasoning should be made clear in the implementation. It should be possible to re-evaluate these argument for any future extension made, we do not want to expand to duplicate `zerocopy` and if it proves necessary that any of this becomes dependent on a user provided type parameter we should switch, imho.

Before the implementation, cc @fintelia Thoughts? I know you lean towards less unsafety here so I'd be interested in knowing which of these options you regard as safer.